lib/spack/spack/bootstrap.py

```diff
@@ -91,6 +91,14 @@ def _try_import_from_store(module, query_spec, query_info=None):
             os.path.join(candidate_spec.prefix, pkg.platlib),
         ]  # type: list[str]
         path_before = list(sys.path)
+
+        # Python 3.8+ on Windows does not search dependent DLLs in PATH,
+        # so we need to manually add it using os.add_dll_directory
+        # https://docs.python.org/3/whatsnew/3.8.html#bpo-36085-whatsnew
+        if sys.version_info[:2] >= (3, 8) and sys.platform == "win32":
```

Comment:
What happens if this is removed?

Reply:
We will get `AttributeError: module 'os' has no attribute 'add_dll_directory'` on (Linux) or (Python 3.7-)

Reply:
Sorry, the comment was referring not the the if statement explicitly but rather the entire stanza.

Reply:
After these lines are removed, Spack bootstrap will fail because it can't `import clingo` correctly. The debug log shows `Warning: spec <a long spec> did not provide module clingo`.

Reply:
Is that from a git ref that includes the recent introduction of rpath support to Spack on Windows?

Reply:
Yes.

Python 3.8+ searches clingo.dll under the directory of clingo.cp310-win_amd64.pyd, which is <prefix>/Lib/site-packages. However, this DLL is actually under <prefix>/bin.

The RPATH support only add DLLs under <prefix>/bin and <prefix>/Lib. Both of them are not searched by Python. Besides, clingo.dll is not symlinked, maybe because this is the build result, not a dependency.

Linux doesn't have this problem, because they have RPATH in the pyd file. If we don't want to solve it by os.add_dll_directory, we may need to symlink that bin/clingo.dll to Lib/site-packages, but I'm not sure if this symlink way is a proper way or not.

Reply:
#33400 resolves this issue using Spack's windows RPath support. RPath on Windows uses symlinking behind the scenes to achieve this, so the symlink methodology you suggested would have been correct.

Reply:
Sorry but I still didn't figure out how to create a symlink from `prefix/bin/clingo.dll` to `prefix/Lib/site-packages`.
I have tried to add `self.win_rpath.add_internal_links(join_path(self.prefix.lib, "site-packages"))` into `patch` but it does not work.